api/app.py

```diff
@@ -0,0 +1,68 @@
+from flask import Flask
+from flask import request
+from flask import url_for
+from geodepy.geodesy import vincdir
+from geodepy.geodesy import vincinv
+from geodepy.convert import dms2dd
+from geodepy.convert import dd2dms
+
+app = Flask(__name__)
+
+coord2dd = {
+    'dd': lambda x: x,
+    'dms': dms2dd,
+}
+
+dd2coord = {
+    'dd': lambda x: x,
+    'dms': dd2dms,
+}
+
+
+@app.route('/')
+def list_routes():
+    return str(tuple(url_for(rule.endpoint) for rule in app.url_map.iter_rules() if rule.endpoint != 'static'))
+
+
+@app.route('/vincinv')
+def handle_vincinv():
+    from_coord = request.args.get('from_coord')
+    to_coord = request.args.get('to_coord')
+    lat1 = request.args.get('lat1', type=float)
+    lon1 = request.args.get('lon1', type=float)
+    lat2 = request.args.get('lat2', type=float)
+    lon2 = request.args.get('lon2', type=float)
+
+    dd = coord2dd[from_coord]
+    lat1_dd, lon1_dd, lat2_dd, lon2_dd = dd(lat1), dd(lon1), dd(lat2), dd(lon2)
+
+    ell_dist_dd, azimuth1to2_dd, azimuth2to1_dd = vincinv(lat1_dd, lon1_dd, lat2_dd, lon2_dd)
+
+    coord = dd2coord[to_coord]
+    ell_dist, azimuth1to2, azimuth2to1 = coord(ell_dist_dd), coord(azimuth1to2_dd), coord(azimuth2to1_dd)
+
+    return str((ell_dist, azimuth1to2, azimuth2to1)), 200
+
+
+@app.route('/vincdir')
```

Comment:
I've been looking for some guidance on whether GET or POST requests make sense for these function calls. On one hand they are (I guess) idempotent which is an argument for GET requests. But on the other hand you're not really GETing a resource - you're asking the server to do something for you, and providing the data which it needs to perform that action. I couldn't find any useful heuristics for this which I trusted, so it seems find how it is.

Reply:
This also goes back to printing out a nice list of endpoints & their arguments. I can't get the arguments in the current method, because the arguments are handled dynamically inside the method.

A more 'static' approach might be to do something like `/vincinv/<lat1>/<lon1>/<lat2>/<lon2>/<from_coord>/<to_coord>`. But I'm really think this makes sense.

Reply:
Path params should be for identifying resources imo.